### PR TITLE
Support Active Directory in a not gross way

### PR DESCRIPTION
**Pull request by jjmason**
Delivers [#101053210].

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -77,11 +77,20 @@ The `conjur-ldap-sync` command accepts the following options:
...
+    --group-object-classes [CLASSES] LDAP objectClasses that should be imported as groups
+    --user-object-classes [CLASSES] LDAP objectClasses that should be imported as users
 
 
 The `--save-api-keys` is off by default, but recommended if you want to allow created roles to login to
 Conjur.
 
+## LDAP Flavors
+
+Active Directory returns a directory structure that differs from posix (OpenLDAP, Apache DS, OpenDJ, and others) structures.  You can tell ldap-sync to use assume Active Directory structures with `--mode active_directory`.  You can also conrol the records that are selected for groups and users with `--user-object-classes oc1,oc2` and `--group-object-classes oc1,oc2`.
```
Comment (kgilpin) [inline README.md]: Can you give examples of real object classes?

File: lib/conjur/ldap/roles.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ module Conjur::Ldap::Roles
   # users.  Their names are preserved as-is, and their uidnumber and gidnumber
   # are set from the LDAP gidNumber and uidNumber.
   # 
-  # Because you can't delete roles Conjur, any roles that are
+  # Because you can't delete roles Conjur, any roles that are``
```
Comment (kgilpin) [inline lib/conjur/ldap/roles.rb]: I think this double-backtick is not meant to be here

File: features/reports.feature [hunk 3]
```diff
@@ -63,4 +64,40 @@ Feature: A report of actions taken is generated
     And I successfully sync
     Then the report should have actions:
       |       tag     |                extra_json                               |
-      | remove_member |  {"group": "<prefix>-admins", "user": "<prefix>-alice"} |
\ No newline at end of file
+      | remove_member |  {"group": "<prefix>-admins", "user": "<prefix>-alice"} |
+
+  Scenario: Text output format
+    Given I initially have an LDAP database with:
```
Comment (kgilpin) [inline features/reports.feature]: Do we have LDIF for AD?

Comment (jjmason) [inline features/reports.feature]: No, at least not in the obvious way.  Ladle (actually ApacheDS) freaks out
out when it sees an AD looking ldif (the ldif that ldapsearch spits out
when you point it at AD).

**Pull request merged**